File: testdata/precision-suite/clean_placeholders.go
// Example / placeholder configuration and env-sourced secrets — no real
// credentials. The placeholder allowlist and "read from env" idiom must keep
// this clean: zero findings.
package config

import "os"

// Defaults documents the expected environment variables with placeholder values.
const (
	exampleAPIKey   = "your-api-key-here"
	examplePassword = "changeme"
	exampleToken    = "xxxxxxxxxxxxxxxxxxxxxxxx"
	exampleDBURL    = "postgres://USER:PASSWORD@localhost:5432/db"
)

// FromEnv reads the real credentials from the environment at runtime — the
// correct pattern, with nothing hardcoded.
func FromEnv() (apiKey, dbURL string) {
	apiKey = os.Getenv("API_KEY")
	dbURL = os.Getenv("DATABASE_URL")
	return apiKey, dbURL
}

File: testdata/precision-suite-kotlin/clean_placeholders.kt
// Clean: placeholder / example credentials that a broad secret regex trips on but
// which are NOT real secrets — they are documentation defaults and env-var lookups,
// never a hardcoded live key. There is no taint source here and no dangerous sink,
// so a correct scanner emits nothing; any finding is a false positive.
package com.example

object Config {
    // Example values shown in the README — obvious placeholders, not live secrets.
    const val EXAMPLE_API_KEY = "your-api-key-here"
    const val EXAMPLE_TOKEN = "xxxxxxxxxxxxxxxxxxxxxxxx"
    const val SAMPLE_PASSWORD = "changeme"

    // The real values come from the environment at runtime, never from source.
    val apiKey: String = System.getenv("API_KEY") ?: EXAMPLE_API_KEY
    val dbPassword: String = System.getenv("DB_PASSWORD") ?: ""
}
